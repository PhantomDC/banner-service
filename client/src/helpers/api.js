import axios from 'axios'
import { baseURL } from '../config'


class API {
  constructor() {
    this.request = axios.create({
      baseURL: `${baseURL}api/`,
    })
  }

  /**
   * сжатие изображение по дефолту
   * @param {*} image
   */
  async compressImage(image) {
    const data = await this.request.post('compress/img', image)

    return data
  }

  /**
   * Сжатие фотографии
   * @param {*} image
   * @param {number} q
   */
  async compressActiveImage(image, q) {
    const data = await this.request.post(`compress/img?quality=${q}`, image)

    return data
  }

  /**
   * Сжатие gif
   * @param {*} image
   * @param {number} q
   */
  async compressGifImage(image, q) {
    const data = await this.request.post(`compress/img?quality=${q}&isGif=1`, image)

    return data
  }

  /**
   * Загрузка архива на сервер
   * @param {*} file
   */
  async uploadArchive(file) {
    const data = await this.request.post('upload', file)

    return data
  }

  /**
   * Загрузка изображения для гиф
   * @param {*} base64
   */
  async uploadImageForGif(base64) {
    const data = await this.request.post('upload/image', base64)

    return data
  }

  /**
   * Генерация гиф
   * @param {*} imgDat
   * @param {*} nameFolde
   */
  async gifGenerated(imgData, nameFolder) {
    const data = await this.request.post('gif/generated', {
      imgData,
      nameFolder,
    })

    return data
  }

  async renameHtmlFile(data) {
    const resolve = await this.request.post('archive/name-update', data)

    return resolve
  }

  /**
   * Добавление границы в файле
   * @param {*} data
   */
  async setBorderFromCanvas(data) {
    const resolve = await this.request.post('update/border', data)

    return resolve
  }

  async getMinimalSize(data) {
    const resolve = await this.request.post('get/minimal-size', data)

    return resolve
  }

  /**
   * Получить информацию о площадках
   */
  async getAreaInfo() {
    const resolve = await this.request.get('area')

    return resolve
  }

  /**
   * Прошить архив
   * @param {*} data
   */
  async firmware(data) {
    const resolve = await this.request.post('firmware', data)

    return resolve
  }
}

export const api = new API()
