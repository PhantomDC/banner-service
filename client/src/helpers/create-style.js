export default (doc, color) => {
  const header = doc.querySelector('head')
  const style = doc.createElement('style')
  let oldStyle

  if (oldStyle = doc.querySelector('style[data-style]')) {
    oldStyle.innerHTML = `body {background-color: ${color} !important}`
    return
  }

  style.setAttribute('data-style', 1)
  style.innerText = `body {background-color: ${color} !important}`
  header.appendChild(style)
}
