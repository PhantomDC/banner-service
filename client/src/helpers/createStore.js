import {
  applyMiddleware, 
  createStore
} from 'redux'
import logger from 'redux-logger'
import thunk from 'redux-thunk'
import { reducers } from '../redux'
import fetch from './middleware/fetch'


export const store = createStore(
  reducers,
  applyMiddleware(
    thunk, logger, fetch
  )
)
