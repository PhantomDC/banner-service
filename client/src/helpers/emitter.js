class EventEmitter {
  constructor() {
    this.events = {}
    this.instance = null
  }

  static getInstance() {
    if (!this.instance) {
      this.instance = new EventEmitter()
      return this.instance
    }
    return this.instance
  }

  on(name, callback) {
    if (!this.events[name]) {
      this.events[name] = callback
    }
  }

  emit(name, ...rest) {
    if (this.events[name]) {
      return this.events[name](...rest)
    }
    throw new Error(`event ${name} not found`)
  }
}

export default new EventEmitter()
