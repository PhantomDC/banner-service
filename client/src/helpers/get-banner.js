export default (playerReady) => {
  if (!playerReady) return false

  const banner = document.getElementById('bannerFrame')
  const doc = banner.contentDocument || banner.contentWindow.document
  const canvas = doc.getElementById('canvas')

  return {
    doc,
    canvas,
  }
}
