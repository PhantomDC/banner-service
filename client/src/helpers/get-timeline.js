export default (root, inst) => {
  if (inst.timeline.duration > 1) {
    return inst.timeline
  }
  return root.timeline
}
