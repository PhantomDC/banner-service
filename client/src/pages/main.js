import React from 'react'
import {
  RootContainer,
  ArchiveBlock,
  CompressImage,
  FlexWrap,
  ShowBannerWithArchive,
  GifImages,
} from '../ui'


export const Main = () => (
  <RootContainer>
    <FlexWrap
      fd="column"
      width="100%"
    >
      <ArchiveBlock />
      <CompressImage />
      <ShowBannerWithArchive />
      <GifImages />
    </FlexWrap>
  </RootContainer>
)
