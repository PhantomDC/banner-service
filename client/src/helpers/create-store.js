import {
  applyMiddleware,
  createStore,
} from 'redux'
import logger from 'redux-logger'
import thunk from 'redux-thunk'
import { reducers } from '../redux'
import fetch from './middleware/fetch'
import nextDispatch from './middleware/next-dispatch'
import { api } from './api'


export const store = createStore(
  reducers,
  applyMiddleware(thunk.withExtraArgument({ api }), logger, fetch, nextDispatch),
)
