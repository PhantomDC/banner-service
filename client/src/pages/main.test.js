import React from 'react'
import { configure, render } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { create } from 'react-test-renderer'
import { wrapProvider } from '../../__TEST__/provider-wrap'
import { store } from '../helpers/create-store'
import { Main } from './main'

/* eslint-disable no-undef */
/* eslint-disable function-paren-newline */

configure({ adapter: new Adapter() })

describe('test page Main', () => {
  it('main page render without crash', () => {
    render(wrapProvider(Main, store))
  })

  it.skip('shapshot Main', () => {
    const tree = create(
      wrapProvider(Main, store),
      {
        createNodeMock: el => ({
          addEventListener() {},
        }),
      },
    )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
