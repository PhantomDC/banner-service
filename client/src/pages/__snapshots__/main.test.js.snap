// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test page Main shapshot Main 1`] = `
<div
  className="sc-htpNat hSPRnb"
>
  <div
    className="sc-bxivhb iZOGDG"
    height="auto"
    width="100%"
  >
    <div
      className="sc-bxivhb bSbxBR"
      height="auto"
      width="auto"
    >
      <div
        className="sc-bxivhb bSbxBR"
        height="auto"
        width="auto"
      >
        <h1
          className="sc-bwzfXH goRTKn"
          size={20}
        >
          Загрузите архив
        </h1>
        <div
          className="sc-bxivhb kDQTAE"
          height="auto"
          width="auto"
        >
          <input
            type="file"
          />
          <button
            className="sc-bdVaJa hvseiO"
            onClick={[Function]}
          >
            Загрузить
          </button>
        </div>
      </div>
      <div>
        <div
          className="sc-bxivhb bSbxBR"
          height="auto"
          style={
            Object {
              "paddingLeft": 0,
            }
          }
          width="auto"
        >
          <p
            className="sc-gqjmRU gZtsxR"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="sc-bZQynM hxnkgJ"
              height="16"
              version="1.1"
              viewBox="0 0 14 16"
              width="14"
            >
              <path
                d="M13 4H7V3c0-.66-.31-1-1-1H1c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h12c.55 0 1-.45 1-1V5c0-.55-.45-1-1-1zM6 4H1V3h5v1z"
                fillRule="evenodd"
              />
            </svg>
            b3f4c6d5859a8e8cf1505a726be42de9--240x400.zip
          </p>
          <div
            className="sc-bxivhb bSbxBR"
            height="auto"
            style={
              Object {
                "paddingLeft": 0,
              }
            }
            width="auto"
          >
            <div
              className="sc-bxivhb bSbxBR"
              height="auto"
              style={
                Object {
                  "paddingLeft": 0,
                }
              }
              width="auto"
            >
              <p
                className="sc-gqjmRU gZtsxR"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="sc-gzVnrw efnvCf"
                  height="16"
                  version="1.1"
                  viewBox="0 0 12 16"
                  width="12"
                >
                  <path
                    d="M6 5H2V4h4v1zM2 8h7V7H2v1zm0 2h7V9H2v1zm0 2h7v-1H2v1zm10-7.5V14c0 .55-.45 1-1 1H1c-.55 0-1-.45-1-1V2c0-.55.45-1 1-1h7.5L12 4.5zM11 5L8 2H1v12h10V5z"
                  />
                </svg>
                240x400.html
              </p>
              <div
                className="sc-bxivhb bSbxBR"
                height="auto"
                style={
                  Object {
                    "paddingLeft": 0,
                  }
                }
                width="auto"
              />
            </div>
            <div
              className="sc-bxivhb bSbxBR"
              height="auto"
              style={
                Object {
                  "paddingLeft": 0,
                }
              }
              width="auto"
            >
              <p
                className="sc-gqjmRU gZtsxR"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="sc-gzVnrw efnvCf"
                  height="16"
                  version="1.1"
                  viewBox="0 0 12 16"
                  width="12"
                >
                  <path
                    d="M6 5H2V4h4v1zM2 8h7V7H2v1zm0 2h7V9H2v1zm0 2h7v-1H2v1zm10-7.5V14c0 .55-.45 1-1 1H1c-.55 0-1-.45-1-1V2c0-.55.45-1 1-1h7.5L12 4.5zM11 5L8 2H1v12h10V5z"
                  />
                </svg>
                240x400.js
              </p>
              <div
                className="sc-bxivhb bSbxBR"
                height="auto"
                style={
                  Object {
                    "paddingLeft": 0,
                  }
                }
                width="auto"
              />
            </div>
            <div
              className="sc-bxivhb bSbxBR"
              height="auto"
              style={
                Object {
                  "paddingLeft": 30,
                }
              }
              width="auto"
            >
              <p
                className="sc-gqjmRU gZtsxR"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="sc-bZQynM hxnkgJ"
                  height="16"
                  version="1.1"
                  viewBox="0 0 14 16"
                  width="14"
                >
                  <path
                    d="M13 4H7V3c0-.66-.31-1-1-1H1c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h12c.55 0 1-.45 1-1V5c0-.55-.45-1-1-1zM6 4H1V3h5v1z"
                    fillRule="evenodd"
                  />
                </svg>
                images
              </p>
              <div
                className="sc-bxivhb bSbxBR"
                height="auto"
                style={
                  Object {
                    "paddingLeft": 30,
                  }
                }
                width="auto"
              >
                <div
                  className="sc-bxivhb bSbxBR"
                  height="auto"
                  style={
                    Object {
                      "paddingLeft": 0,
                    }
                  }
                  width="auto"
                >
                  <p
                    className="sc-gqjmRU gZtsxR"
                    onClick={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="sc-gzVnrw efnvCf"
                      height="16"
                      version="1.1"
                      viewBox="0 0 12 16"
                      width="12"
                    >
                      <path
                        d="M6 5H2V4h4v1zM2 8h7V7H2v1zm0 2h7V9H2v1zm0 2h7v-1H2v1zm10-7.5V14c0 .55-.45 1-1 1H1c-.55 0-1-.45-1-1V2c0-.55.45-1 1-1h7.5L12 4.5zM11 5L8 2H1v12h10V5z"
                      />
                    </svg>
                    kn.png
                  </p>
                  <div
                    className="sc-bxivhb bSbxBR"
                    height="auto"
                    style={
                      Object {
                        "paddingLeft": 0,
                      }
                    }
                    width="auto"
                  />
                </div>
                <div
                  className="sc-bxivhb bSbxBR"
                  height="auto"
                  style={
                    Object {
                      "paddingLeft": 0,
                    }
                  }
                  width="auto"
                >
                  <p
                    className="sc-gqjmRU gZtsxR"
                    onClick={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="sc-gzVnrw efnvCf"
                      height="16"
                      version="1.1"
                      viewBox="0 0 12 16"
                      width="12"
                    >
                      <path
                        d="M6 5H2V4h4v1zM2 8h7V7H2v1zm0 2h7V9H2v1zm0 2h7v-1H2v1zm10-7.5V14c0 .55-.45 1-1 1H1c-.55 0-1-.45-1-1V2c0-.55.45-1 1-1h7.5L12 4.5zM11 5L8 2H1v12h10V5z"
                      />
                    </svg>
                    pic2.jpg
                  </p>
                  <div
                    className="sc-bxivhb bSbxBR"
                    height="auto"
                    style={
                      Object {
                        "paddingLeft": 0,
                      }
                    }
                    width="auto"
                  />
                </div>
                <div
                  className="sc-bxivhb bSbxBR"
                  height="auto"
                  style={
                    Object {
                      "paddingLeft": 0,
                    }
                  }
                  width="auto"
                >
                  <p
                    className="sc-gqjmRU gZtsxR"
                    onClick={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="sc-gzVnrw efnvCf"
                      height="16"
                      version="1.1"
                      viewBox="0 0 12 16"
                      width="12"
                    >
                      <path
                        d="M6 5H2V4h4v1zM2 8h7V7H2v1zm0 2h7V9H2v1zm0 2h7v-1H2v1zm10-7.5V14c0 .55-.45 1-1 1H1c-.55 0-1-.45-1-1V2c0-.55.45-1 1-1h7.5L12 4.5zM11 5L8 2H1v12h10V5z"
                      />
                    </svg>
                    pic4.jpg
                  </p>
                  <div
                    className="sc-bxivhb bSbxBR"
                    height="auto"
                    style={
                      Object {
                        "paddingLeft": 0,
                      }
                    }
                    width="auto"
                  />
                </div>
                <div
                  className="sc-bxivhb bSbxBR"
                  height="auto"
                  style={
                    Object {
                      "paddingLeft": 0,
                    }
                  }
                  width="auto"
                >
                  <p
                    className="sc-gqjmRU gZtsxR"
                    onClick={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="sc-gzVnrw efnvCf"
                      height="16"
                      version="1.1"
                      viewBox="0 0 12 16"
                      width="12"
                    >
                      <path
                        d="M6 5H2V4h4v1zM2 8h7V7H2v1zm0 2h7V9H2v1zm0 2h7v-1H2v1zm10-7.5V14c0 .55-.45 1-1 1H1c-.55 0-1-.45-1-1V2c0-.55.45-1 1-1h7.5L12 4.5zM11 5L8 2H1v12h10V5z"
                      />
                    </svg>
                    pic5.jpg
                  </p>
                  <div
                    className="sc-bxivhb bSbxBR"
                    height="auto"
                    style={
                      Object {
                        "paddingLeft": 0,
                      }
                    }
                    width="auto"
                  />
                </div>
                <div
                  className="sc-bxivhb bSbxBR"
                  height="auto"
                  style={
                    Object {
                      "paddingLeft": 0,
                    }
                  }
                  width="auto"
                >
                  <p
                    className="sc-gqjmRU gZtsxR"
                    onClick={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="sc-gzVnrw efnvCf"
                      height="16"
                      version="1.1"
                      viewBox="0 0 12 16"
                      width="12"
                    >
                      <path
                        d="M6 5H2V4h4v1zM2 8h7V7H2v1zm0 2h7V9H2v1zm0 2h7v-1H2v1zm10-7.5V14c0 .55-.45 1-1 1H1c-.55 0-1-.45-1-1V2c0-.55.45-1 1-1h7.5L12 4.5zM11 5L8 2H1v12h10V5z"
                      />
                    </svg>
                    te.png
                  </p>
                  <div
                    className="sc-bxivhb bSbxBR"
                    height="auto"
                    style={
                      Object {
                        "paddingLeft": 0,
                      }
                    }
                    width="auto"
                  />
                </div>
                <div
                  className="sc-bxivhb bSbxBR"
                  height="auto"
                  style={
                    Object {
                      "paddingLeft": 0,
                    }
                  }
                  width="auto"
                >
                  <p
                    className="sc-gqjmRU gZtsxR"
                    onClick={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="sc-gzVnrw efnvCf"
                      height="16"
                      version="1.1"
                      viewBox="0 0 12 16"
                      width="12"
                    >
                      <path
                        d="M6 5H2V4h4v1zM2 8h7V7H2v1zm0 2h7V9H2v1zm0 2h7v-1H2v1zm10-7.5V14c0 .55-.45 1-1 1H1c-.55 0-1-.45-1-1V2c0-.55.45-1 1-1h7.5L12 4.5zM11 5L8 2H1v12h10V5z"
                      />
                    </svg>
                    tuman.jpg
                  </p>
                  <div
                    className="sc-bxivhb bSbxBR"
                    height="auto"
                    style={
                      Object {
                        "paddingLeft": 0,
                      }
                    }
                    width="auto"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bxivhb iZOGDG"
      height="auto"
      width="100%"
    >
      <div
        className="sc-bxivhb iZOGDG"
        height="auto"
        style={
          Object {
            "visibility": "hidden",
          }
        }
        width="100%"
      >
        <div
          className="sc-VigVT eiTpYg"
          height="auto"
          width="auto"
        >
          <div
            className="sc-iwsKbI fKjEyi"
            onClick={[Function]}
          />
          <div
            className="sc-jTzLTM gPRvrq"
            height="auto"
            style={
              Object {
                "transform": "translateX(0px)",
              }
            }
            width="0px"
          />
          <div
            className="sc-iwsKbI dBtNkJ"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-bxivhb hsPTVF"
        height="auto"
        width="100%"
      />
    </div>
    <div
      className="sc-kpOJdX ctKjuL"
      height="auto"
      width="100%"
    >
      <div
        className="sc-bxivhb kDQTAE"
        height="auto"
        width="auto"
      >
        <button
          className="sc-bdVaJa hvseiO"
          onClick={[Function]}
        >
          посмотреть баннер
        </button>
      </div>
    </div>
    <div
      className="sc-bxivhb hsPTVF"
      height="auto"
      width="100%"
    />
  </div>
</div>
`;
